feat(migration-docs): Add Stardust models

lzpap opened the pull request:
# Description of change

 - Migrated stardust migration from "advanced topics" to dedicated guide section (`guides/stardust/migration-process`), to be imporved in follow up PR
 - Added `stardust` and `timelock` package documentation to the references on the docs site
 - Disabled `deepbook` package docs
 - Added stardust models page

To see the new page(s):

```bash
rm -rf docs/content/references/framework
cd docs/site
pnpm build
pnpm start
```

Open the sites:
 - http://localhost:3000/guides/stardust-migration
 - http://localhost:3000/guides/stardust/move-models
 - http://localhost:3000/references/framework

## Links to any relevant issues
Fixes #1041

lzpap committed:
feat(docs): move stardust migration doc

lzpap committed:
feat(docs): add `stardust` and `timelock` package reference

lucas-tortora committed:
fix: add slugs manually to avoid the category page default behaviour

lzpap committed:
Merge pull request #905 from iotaledger/devEx/fixFrameworkBuildErrors

fix (migration docs): Fix Build Errors

lzpap committed:
feat(docs): Add stardust models

lzpap committed:
Merge branch 'develop' into sc-platform/docs/stardust-models

lucas-tortora reviewed (changes requested):
Most of these are minor stuff, I can create a PR if it's easier.

lucas-tortora committed:
capitalized Stardust
Updated broken links
Changed GitHub tip URLs to Wiki
fixed title in Foundries Output
minor edits

lucas-tortora committed:
changed links to tips from Wiki to GH

lzpap committed:
Merge pull request #975 from iotaledger/devex/edit/sc-platform/docs/stardust-models

Devex - Docs - Edit feat(migration-docs): Add Stardust models #854

lzpap committed:
Merge branch 'develop' into sc-platform/docs/stardust-models
